src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -4397,18 +4324,21 @@ def _contains_intrinsic_reduction(pnodes):
         # parent for this logical expression we will repeat the processing.
         fake_parent = Assignment(parent=parent)
         self.process_nodes(fake_parent, logical_expr)
+        # We want to convert all the non-explicit array syntax code to use
+        # explicit array syntax.
+        # TODO 1799 If we have two arrays where one uses a non-maximal range
+        # that is defined we will not convert this correctly yet.
+        # Convert References to arrays to use the array range notation unless
+        # they have an IntrinsicCall parent.
+        for ref in fake_parent.walk(Reference):
+            if (isinstance(ref.symbol, DataSymbol) and
+                    not ref.ancestor(IntrinsicCall)):
+                try:
+                    Reference2ArrayRangeTrans().apply(ref)
+                except TransformationError:
+                    pass
```

Comment:
Is it safe to ignore the TransformationErrors? I see one is related to intrinsics that may be why you need to ignore them before the try?

Reply:
I think so - as far as I can see that all the TransformationErrors come without any modifications to the tree - if the transformation fails on an array, then in the worst case we get the current status (i.e. a CodeBlock).

Reply:
The issue may not come exactly from here, but if I do a test with:
```
    where(a(:) + b > 1)
        b = c + d
    end where
```
where all a, b, c, and d are improrted (so we don't know the types), it converts it to:
```
  do widx1 = 1, SIZE(a, dim=1), 1
    if (a(LBOUND(a, dim=1) + widx1 - 1) + b > 1) then
      b = c + d
    end if
  enddo
``` 
Which misses the indices in the condition 'b' and all the indices in the statement. The generated code also doesn't compile.

Reply:
Fixed this now I think.